polynote-frontend/polynote/ui/component/notebook/kernel.ts

```diff
@@ -50,11 +50,13 @@ export class KernelPane extends Disposable {
                 // the notebook should already be loaded
                 if (nbInfo?.info) {
                     if (this.kernels[path] === undefined) {
-                        this.kernels[path] = new Kernel(
+                        const kernel = new Kernel(
```

Comment:
This was the problem FWIW – the `Kernel` instance was hanging around after the notebook was closed, so it was in a bad state and wouldn't get reinitialized.